shm/Cargo.hpp

```diff
@@ -7,9 +7,10 @@ class Cargo {
     Cargo(const std::string& name, size_t amount, size_t basePrice)
         : name_(name), amount_(amount), basePrice_(basePrice) {}
 
-    const std::string& getName() const { return name_; }
-    size_t getAmount() const { return amount_; }
-    size_t getBasePrice() const { return basePrice_; }
+    virtual std::string getName() const = 0;
+    virtual size_t getAmount() const = 0;
+    virtual size_t getBasePrice() const = 0;
```

Comment:
These functions do not have to be virtual and can be implemented in Cargo class. Every other class inheriting from it will have access to them. There is no point in redefining getters for name, amount and base price while all of these variables will be present across every class derived from Cargo

Reply:
This is based on Task1 from https://github.com/coders-school/object-oriented-programming/blob/master/module2/presentation_virtual.md

> Przekształć klasę Cargo w interfejs z 4 czysto wirtualnymi metodami.

Reply:
I've seen you asked Mateusz so let's wait for his answer. For now I'll leave it like this.

Reply:
So it's to be left as is due to educational value following Mateusz's response 

> @Kuba C Tak, ale to jest po to abyście się pouczyli pisać interface :slight_smile: Możliwe, że z czasem zmieńcie albo dodacie kolejne klasy dziedzicząc epo Cargo i będziecie chcieli zmienić ich zachowania i wtedy dzięki interface to będzie możliwe. A tak na dzień dobry wymuszacie jedno rozwiązanie. Przykładowo getName() dla nowej klasy Wood, może sprawdzać typ drewna i wypisywać konkretną nazwę np. Oak, a nie zawsze Wood.